src/lib/ppx_version/bin_io_unversioned.ml

```diff
@@ -91,21 +91,22 @@ let ctxt_base =
     ~input_name:""
 
 let rewrite_to_bin_io ~loc ~path (_rec_flag, type_decls) =
-  let type_decl1 = List.hd_exn type_decls in
-  if not (Int.equal (List.length type_decls) 1) then
-    Location.raise_errorf ~loc
-      "deriving bin_io_unversioned can only be used on a single type" ;
-  let module_path = List.drop String.(split path ~on:'.') 2 in
-  let inner2_modules = List.take (List.rev module_path) 2 in
-  validate_type_decl inner2_modules type_decl1 ;
-  let ctxt =
-    let derived_item_loc = loc in
-    (* TODO: is inline:false what we want? *)
-    Expansion_context.Deriver.make ~derived_item_loc ~base:ctxt_base ()
-      ~inline:false
-  in
-  List.concat_map bin_io_gens ~f:(fun gen ->
-      gen ~ctxt (Nonrecursive, type_decls) [] )
+  match type_decls with
+  | [ type_decl1 ] ->
+      let module_path = List.drop String.(split path ~on:'.') 2 in
+      let inner2_modules = List.take (List.rev module_path) 2 in
+      validate_type_decl inner2_modules type_decl1 ;
+      let ctxt =
+        let derived_item_loc = loc in
+        (* TODO: is inline:false what we want? *)
+        Expansion_context.Deriver.make ~derived_item_loc ~base:ctxt_base ()
+          ~inline:false
+      in
+      List.concat_map bin_io_gens ~f:(fun gen ->
+          gen ~ctxt (Nonrecursive, type_decls) [] )
+  | [] | _ :: _ :: _ ->
```

Comment:
I think the empty list is impossible, so `assert false` would be appropriate.

Reply:
Done